dali/test/python/test_operator_python_function.py

```diff
@@ -486,8 +485,44 @@ def __init__(self, batch_size, num_threads, device_id, _seed):
     def define_graph(self):
         return self.op()
 
-
 @raises(RuntimeError)
 def test_wrong_pipeline():
     pipe = AsyncPipeline(BATCH_SIZE, NUM_WORKERS, DEVICE_ID, SEED)
     pipe.build()
+
+def test_output_layout():
+    pipe = CommonPipeline(1, 1, 0, 999, images_dir)
+    with pipe:
+        images, _ = pipe.load()
+        out1, out2 = fn.python_function(images,
+                                        function=lambda x: (x, x.mean(2)),
+                                        num_outputs=2,
+                                        output_layouts=['HWC', 'HW'])
+        out3, out4 = fn.python_function(images, 
+                                        function=lambda x: (x, x/2),
+                                        num_outputs=2,
+                                        output_layouts='HWC')
+        out5, out6 = fn.python_function(images,
+                                        function=lambda x: (x, x/2),
+                                        num_outputs=2,
+                                        output_layouts=['HWC'])
+
+        pipe.set_outputs(out1, out2, out3, out4, out5, out6)
+    pipe.build()
+    out1, out2, out3, out4, out5, out6 = pipe.run()
+    assert(out1.layout() == 'HWC')
```

Comment:
Maybe you can use custom layouts, and distinct for each case, like `ABC` or something.

Reply:
done